# HapMLST

HapMLST is a pipeline that processes metagenomic data using EM algorithms and applies the allele sequence variants to quickly and accurately assess the abundance and ST type of bacteria based MLST classification.

# Pipeline


## Installation
### 1. install with docker??
- step1: ...
- step2: ...
### 2. install with Source code
- step1: git clone https://github.com/Chen-Lab123/HapMLST.git
- step2: 

## Database Preparation
All MLST classifications are based on PubMLST Database (before 2023.8.31): https://pubmlst.org/. 

Prebuild database of xxx bacteria species can be downloaded from Zenodo (doi:xxx). 

IN Zenodo, each tar.gz file represents a species, in following format: species_name-version_number.tar.gz (eg. in Escherichia-coli-202308.tar.gz, Escherichia-coli is the species_name and 202308 is the version_number)

Before running HapMLST, make sure the tar.gz of the target species is already unzipped within the database folder (xxx):

```
cd xxx
wget xxx
tar -zxvf Escherichia-coli-202308.tar.gz
```

Each species has above file:

- file 1: xxx;
- file 2: xxx;

For new species, user-specific database could be generated by the following way:
- step 1: ...;
- step 2: ...;
- step 3: ...;

## Run Analysis
Downlaod the test data into folder (xxx). Test data is in Zenodo (doi:xxx), which is a simulation dataset with xxx samples from _E. coli_ genome.

Note: the input files could be 1 or 2 with .fq, .fastq or .fq.gz, .fastq.gz. 

## Citation

## FAQs
