# HapMLST

HapMLST is a pipeline that processes metagenomic data using EM algorithms and applies the allele sequence variants to quickly and accurately assess the abundance and ST type of bacteria based MLST classification.

## Pipeline
![image](https://github.com/Chen-Lab123/HapMLST/blob/main/pipeline/Figure1-1.png)

## Installation
### 1. install with docker??
- step1: ...
- step2: ...
### 2. install with Source code
- step1: git clone https://github.com/Chen-Lab123/HapMLST.git
- step2: 

## Database Preparation
All MLST classifications are summarized by tseemann basing on PubMLST Database.  You can download from : https://github.com/tseemann/mlst/tree/master/db.

If you want to more updated MLST information, please try to search in PubMLST Database: https://pubmlst.org/.


## Run Analysis
Downlaod the test data into folder (xxx). Test data is in Zenodo (doi:xxx), which is a simulation dataset with xxx samples from _E. coli_ genome.

Note: the input files could be 1 or 2 with .fq, .fastq or .fq.gz, .fastq.gz. 

## Citation

## FAQs
